src/diffpy/utils/resampler.py

```diff
@@ -79,6 +79,11 @@ def wsinterp(x, xp, fp, left=None, right=None):
     return fp_at_x
 
 
+@deprecated(
+    "The 'resample' function is deprecated and will be removed in a future release 3.6.3. \n"
+    "'resample' has been renamed 'wsinterp' to better reflect functionality. \n"
+    "Please use 'wsinterp' instead."
+)
```

Comment:
I'll test on that and make a new PR based on the new structure

Reply:
@bobleesj this deprecated decorator actually is only released for 3.13 so should I use warnings.warn instead? @Sparks29032

Reply:
Yes, it's a new feature for 3.13. 

If it doesn't work for 3.11 and 3.12, then I suggest we should not use this decorator since then 3.11 and 3.12 users won't be able to read this warning msg. tagging @sbillinge for his opinion too.